app/controllers/idv/sessions_controller.rb

```diff
@@ -1,24 +1,55 @@
 module Idv
+  # NOTE: This controller is used by both IDV and Inherited Proofing (IP). The IP
+  # proofing process is "inserted" into the middle of the IDV proofing process and
+  # uses its own IP Flow-specific session; consequently, when the IP proofing
+  # process is underway, any session clean up that needs to take place, MUST involve
+  # both the IDV AND IP sessions. In the case of the IDV proofing process, only
+  # the IDV session needs to be cleaned up.
+  #
+  # Currently, the only action in this controller is the :destroy action; if additional
+  # actions are added to this controller, please filter the before_action callbacks
+  # appropariately (e.g. only: [:destroy], etc.).
   class SessionsController < ApplicationController
```

Comment:
I feel like that implies that when starting over, we don't know where we are? At the time we show a "start over" link, we should know if there is VAIP data in the session, or if there's just regular IDV data right? ex that's why we render `idv/inherited_proofing/cancel` instead of shared cancel?

Reply:
We know where we are, but there is session data for both IDV _and_ IP when IP is in play. Depending on where we are in the IP process, we could be canceling from either an IP flow step or an IDV flow step. That is neither here nor there, but whatever happens in whatever SessionController we call, we need to clean up both IP _and_ IDV session data, the approach I chose handles that with, what I believe, is the most straight forward approach. No matter how many Services Providers the IP process takes on, that code will never change. It's just an attempt to make sure both IDV _and_ IP session are cleaned up properly so the user can start from scratch no matter where they are in the IP process.

Reply:
With the IDV work flow, it looks like this: IDV -> IDV.
The IP workflow looks like this: IDV -> IP -> IDV. Users can cancel anywhere within that < process. They can be in an IP flow step, they can be in an IDV flow step, but when IP is in play, that is what the IP workflow consists of.

Reply:
@zachmargolis I'm still writing up specs for this, but do you have any more questions/thoughts on this?

Reply:
I got a little busy, I'm sure I'll have more thoughts but I need to circle back to this later

Reply:
There may be a chance to use 2 controllers now that I think of it, but I have to make sure that after the IP session data is passed along to IDV, the IP session data is cleaned up at that point. Once the IP process hands off to the IDV, the only way we can go back to the IP process is if the user "Start[s] Over", in which case the IP session data should start clean, so I don't believe there is any reason to hand on to it at that point, it will just get re-populated by the IP process and start over. I'll have a look.